weave

```diff
@@ -1478,6 +1564,13 @@ case "$COMMAND" in
                 --rewrite-hosts)
                     REWRITE_HOSTS=1
                     ;;
+                --add-host)
+                    DNS_EXTRA_HOSTS="$2 $DNS_EXTRA_HOSTS"
+                    shift
+                    ;;
+                --add-host=*)
+                    DNS_EXTRA_HOSTS="${1#*=} $DNS_EXTRA_HOSTS"
+                    ;;
```

Comment:
I wonder if `dns_extra_hosts_args` would be extended to handle `--rewrite-hosts`; this would simplify the `run` implementation too.  And then you might throw in `--or-die` too.